wrangler-core/src/main/java/io/cdap/directives/parser/FixedLengthParser.java

```diff
@@ -94,16 +94,20 @@ public void destroy() {
       int idx = row.find(col);
       if (idx != -1) {
         Object object = row.getValue(idx);
+
+        if (object == null) {
+          throw new DirectiveExecutionException(
+            NAME, String.format("Column '%s' has null value. It should be a non-null 'String'.", col));
+        }
+
         if (object instanceof String) {
           String data = (String) object;
           int length = data.length();
           // If the recordLength length doesn't match the string length.
           if (length < recordLength) {
             throw new ErrorRowException(
-              String.format("Fewer bytes than length of row specified - expected atleast %d bytes, found %s bytes.",
-                            recordLength, length),
-              2
-            );
+              NAME, String.format("Fewer bytes than length of row specified - expected atleast %d bytes, " +
```

Comment:
should include the column name here. Let's also fix this to have a space in between atleast and to accurately talk about number of characters and not bytes:

Column '%s' contains a value with fewer characters than the specified length of row. Expected at least %d characters but found %s characters.